test_e2e.sh

```diff
@@ -621,6 +624,52 @@ remove_sync_container
 wait
 pass
 
+##############################################
+# Test password
+##############################################
+testcase "password"
+echo "$TESTCASE 1" > "$REPO"/file
+git -C "$REPO" commit -qam "$TESTCASE 1"
+# run with askpass_git but with wrong password
+GIT_SYNC \
+    --git=$ASKPASS_GIT \
+    --username="you@example.com" \
+    --password="I have no idea what the password is." \
+    --logtostderr \
+    --v=5 \
+    --wait=0.1 \
+    --repo="file://$REPO" \
+    --branch=master \
+    --rev=HEAD \
+    --root="$ROOT" \
+    --dest="link" \
+    > "$DIR"/log."$TESTCASE" 2>&1 &
```

Comment:
You can change this to not run in the background and not sleep if you switch to `--one-time`